### PR TITLE
[REEF-241]: cleanup clean target for c#/cpp projects

### PR DESCRIPTION
**Pull request by beysims**
This addressed the issue by deleting target directories and moving intermediate files to bin directory

JIRA: [REEF-241](https://issues.apache.org/jira/browse/REEF-241)

This closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj [hunk 1]
```diff
@@ -23,7 +23,6 @@ under the License.
     <RootNamespace>Org.Apache.REEF.Bridge.JAR</RootNamespace>
     <ProjectGuid>{62905C7C-1A7E-4923-B78D-1BF42D7FAD40}</ProjectGuid>
     <AssemblyName>Org.Apache.REEF.Bridge.JAR</AssemblyName>
-	<OutputType>JAR</OutputType>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj]: What was it used for?

File: lang/cs/build.props [hunk 2]
```diff
@@ -133,4 +134,10 @@ under the License.
       DestinationFiles="@(MySourceFiles->'$(TargetDir)%(Filename)%(Extension)')"
     />
   </Target>
+    <ItemGroup>
+        <DirectoriesToRemove Include="$(IntermediateOutputPath)" />
+     </ItemGroup>
+     <Target Name="RemoveIntermediateDirectories"  AfterTargets="clean">
+        <RemoveDir Directories="@(DirectoriesToRemove)" Condition="!Exists('$(IntermediateOutputPath)\$(RootNamespace).log')"/>
```
Comment (jwang98052) [inline lang/cs/build.props]: What is the condition used for?

File: lang/cs/build.props [hunk 3]
```diff
@@ -133,4 +134,10 @@ under the License.
       DestinationFiles="@(MySourceFiles->'$(TargetDir)%(Filename)%(Extension)')"
     />
   </Target>
+    <ItemGroup>
+        <DirectoriesToRemove Include="$(IntermediateOutputPath)" />
+     </ItemGroup>
+     <Target Name="RemoveIntermediateDirectories"  AfterTargets="clean">
```
Comment (jwang98052) [inline lang/cs/build.props]: Where does RemoveIntermediateDirectories target get called? Is it by default?

Comment (jwang98052) [inline lang/cs/build.props]: AfterTargets="clean" should clean be Clean? not sure if it is case sensitive.

File: lang/cs/build.props (hunk 2 above)
Comment (jwang98052) [inline lang/cs/build.props]: It cleans temp folder. Not dlls. Shall we define Clean target to clean $(OutputPath)?

File: lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj (hunk 1 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj]: I don't think it matters. In fact, it only seems to confuse Visual Studio :-)

File: lang/cs/build.props (hunk 2 above)
Comment (markusweimer) [inline lang/cs/build.props]: I'd be in favor of nuking the whole `$(OutputPath)`. That way, the `bin` folder serves the same function as the `target` folder in Maven. We can also put our generated sources there at a later time.

File: lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj (hunk 1 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj]: That being said, the change is unrelated to the purpose of this PR and it would be good hygiene to remove it before merging.

File: lang/cs/build.props [hunk 4]
```diff
@@ -133,4 +134,10 @@ under the License.
       DestinationFiles="@(MySourceFiles->'$(TargetDir)%(Filename)%(Extension)')"
     />
   </Target>
+    <ItemGroup>
```
Comment (markusweimer) [inline lang/cs/build.props]: The indent seems off from the rest of the file.
